### PR TITLE
Feature/sva 118 get single project

### PR DESCRIPTION
**Pull request by stevenjburrows**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Checklist](#checklist)

# Description

added an endpoint to get a single project
The endpoint is localhost:3000/projects/single?res=15486&it=IT-347
res is taken from res_id and it can be taken from it_key both are in the data returned from /projects

Fixes # (issue number)

## Type of change

Please delete options that are not relevant
- [] Bug fix (non-breaking change which fixes an isssue)
- [x] New Feature (non-breaking change which adds functionlity)
- [] Breaking change(fix or feature that would cause existing functionlity to not work as expected)
- [] This change requires a documentation update

# Checklist

- [] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [] I have commented my code, pariculary in hard-to-understand areas
- [] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by CatMac228 — APPROVED** [review]
Approved - data returned: 
image

File: api/routes/projects.js [hunk 1]
```diff
@@ -122,4 +122,73 @@ router.get('/', async (req, res, next) => {
   });
 });
 
+router.get('/single', async (req, res, next) => {
+
+  const singleRes = await axios.get(
```
Comment (joefhall) [inline api/routes/projects.js:127]: Is there a reason why the URL accepts both `res` and `it` parameters?  Could it instead only accept an `it` parameter and then the code does the job of fetching only the resources relevant to that project?  (Also then you could use a classic API URL format of `/projects/{project ID}` e.g. `/projects/IT-347`)

File: api/routes/projects.js [hunk 2]
```diff
@@ -122,4 +122,73 @@ router.get('/', async (req, res, next) => {
...
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          // below use email address you used for jira and generate token from jira
+          `${email}:${api_key}`,
+        ).toString('base64')}`,
+        Accept: 'application/json',
+      },
+    },
+  )
+  // .then((resData) => {
```
Comment (joefhall) [inline api/routes/projects.js:139]: Could the commented out lines here and below be deleted?

File: api/routes/projects.js [hunk 3]
```diff
@@ -122,4 +122,73 @@ router.get('/', async (req, res, next) => {
...
+  //   res.status(200).json(project)
+
+ const [resResults, itResults] = await Promise.all([singleRes, singleIt])
+  const project = {
+        res_id: resResults.data.id,
+        it_related_field_id: resResults.data.fields.customfield_10109,
+        jobRole: resResults.data.fields.customfield_10113,
+        projectType: resResults.data.fields.customfield_10112,
+        suitableForBuddy: resResults.data.fields.customfield_10108 ? resResults.data.fields.customfield_10108.value : 'none',
+        candidateTime: resResults.data.fields.customfield_10062 ? resResults.data.fields.customfield_10062 : 'none',
```
Comment (joefhall) [inline api/routes/projects.js:181]: If you wanted to, I think you could simplify this and some of the other lines below that follow a similar format, by using a [nullish coalesce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) e.g. `candidateTime: resResults.data.fields.customfield_10062 ?? : 'none',`

**Comment by stevenjburrows** [comment]
@joefhall removed the commented out code and edited the test comment.  I tried the nullish link but I got errors.

File: api/routes/projects.js (hunk 3 above)
Comment (joefhall) [reply api/routes/projects.js:160]: Sorry, just realised there was a typo in my comment -- that could be what was causing you problems!  I think `candidateTime: resResults.data.fields.customfield_10062 ?? 'none',` should work? (I'd left in a stray `:` after  the`??` -- my bad 🤦‍♂️ )

**Review by joefhall — APPROVED** [review]

**Pull request merged**